### PR TITLE
New machines marked as NOT_WORKING

### PR DESCRIPTION
**Pull request by clawgrip**
----------------------------------
Diamond King [jordigahan, ClawGrip]

### CODE REVIEW

File: src/mame/drivers/diamondking.cpp [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW3:8")
+INPUT_PORTS_END
+
+void diamondking_state::diamondking(machine_config &config)
+{
+	I8085A(config, m_maincpu, 6.144_MHz_XTAL);
+
+	SPEAKER(config, "mono").front_center();
+
+	ay8910_device &ay8910(AY8910(config, "ay8910", XTAL(2'000'000))); // Frecuency unknown
```
Comment (cuavas) [inline src/mame/drivers/diamondking.cpp:155]: Don’t use the `XTAL` macro when you’re just putting in a guessed frequency.  Also, spelling.  Same applies for the 68340 and MSM6585

Comment (clawgrip) [reply src/mame/drivers/diamondking.cpp]: Fixed, thanks!!

File: src/mame/drivers/diamondking.cpp [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+	SPEAKER(config, "mono").front_center();
+
+	ay8910_device &ay8910(AY8910(config, "ay8910", 2'000'000)); // Frecuency unknown
+	ay8910.port_a_read_callback().set_ioport("DSW1");
+	ay8910.add_route(ALL_OUTPUTS, "mono", 1.0); // Guess
+
+	M68340(config, m_iocpu, 16'000'000); // Frecuency unknown
+
+	msm6585_device &msm6585(MSM6585(config, "msm6585", 0'640'000)); // Frecuency unknown
```
Comment (cuavas) [inline src/mame/drivers/diamondking.cpp:161]: You’ve made an octal constant - that's about 213kHz now, which is probably not what you meant.

**Review by cuavas — CHANGES_REQUESTED** [review]

File: src/mame/drivers/diamondking.cpp (hunk 2 above)
Comment (happppp) [reply src/mame/drivers/diamondking.cpp:161]: correct spelling is with a Q, frequency

Comment (clawgrip) [reply src/mame/drivers/diamondking.cpp]: Fix spelling and OKI speed, thanks a lot.

**Pull request merged**